ops/model.py

```diff
@@ -1937,8 +1940,22 @@ def from_name(cls, name: str, message: str):
             return cls._statuses[typing.cast(StatusName, name)](message)
 
     @classmethod
-    def register(cls, child: Type['StatusBase']):
-        """Register a Status for the child's name."""
+    def register(cls, child: Type[_StatusClass]) -> Type[_StatusClass]:
+        """Class decorator to register a subclass for lookup using :meth:`from_name`.
+
+        Note: this method is intended for internal use only.
+        It is used to make the six valid Juju statuses available by name.
```

Comment:
I would drop the "six" - there's no current intention to add a 7th as far as I know, but if that does happen then it seems very likely that the "six" here would get missed when we added support since nothing else in this method would need to change. I don't the number is particularly important in the doc.

```suggestion
        It is used to make the valid Juju statuses available by name.
```